Tests/AppTests/Controllers/EventSpeechControllerTests.swift

```diff
@@ -0,0 +1,229 @@
+    XCTAssertEqual(contentJSON?["title"]?.string, content.title)
+    XCTAssertEqual(contentJSON?["description"]?.string, content.description)
+    XCTAssertEqual(contentJSON?["link"]?.string, content.link)
+    XCTAssertEqual(contentJSON?["type"]?.string, content.type.string)
+  }
+  
+  func testThatIndexSpeechesReturnsCorrectSpeechesCount() throws {
+    guard let eventId = try storeEvent(), let id = eventId.int else {
+      XCTFail()
+      return
+    }
+    
+    let expectedSpeechesCount = Int.random(min: 1, max: 6)
+    for _ in 0..<expectedSpeechesCount {
+      try storeSpeech(forEventId: eventId)
+    }
+    
+    let res = try fetchSpeeches(parameterId: id)
+    let arrayJSON = res.json?.array
+    
+    XCTAssertEqual(arrayJSON?.count, expectedSpeechesCount)
+  }
+  
+  func testThatIndexSpeechesReturnsCorrectSpeakersCount() throws {
+    guard let eventId = try storeEvent(), let id = eventId.int else {
+      XCTFail()
+      return
+    }
+    
+    let expectedSpeakersCount = Int.random(min: 1, max: 6)
+    try storeSpeech(forEventId: eventId, speakersCount: expectedSpeakersCount)
+    
+    let res = try fetchSpeeches(parameterId: id)
+    let speechJSON = res.json?.array?.first
+    let speakersArrayJSON = speechJSON?["speakers"]?.array
+    
+    XCTAssertEqual(speakersArrayJSON?.count, expectedSpeakersCount)
+  }
+  
+  func testThatIndexSpeechesReturnsCorrectContentsCount() throws {
+    guard let eventId = try storeEvent(), let id = eventId.int else {
+      XCTFail()
+      return
+    }
+    
+    let expectedContentsCount = Int.random(min: 1, max: 6)
+    try storeSpeech(forEventId: eventId, contentCount: expectedContentsCount)
+    
+    let res = try fetchSpeeches(parameterId: id)
+    let speechJSON = res.json?.array?.first
+    let contentsArrayJSON = speechJSON?["contents"]?.array
+    
+    XCTAssertEqual(contentsArrayJSON?.count, expectedContentsCount)
+  }
+    
+  // MARK: Endpoint tests
+  
+  func testThatGetSpeechesRouteReturnsOkStatus() throws {
+    guard let eventId = try storeEvent(), let id = eventId.int else {
+      XCTFail()
+      return
+    }
+    
+    try storeSpeech(forEventId: eventId)
+    
+    try drop
+      .clientAuthorizedTestResponse(to: .get, at: "/event/\(id)/speech")
+      .assertStatus(is: .ok)
+  }
+}
+
+extension EventSpeechControllerTests {
+  
+  fileprivate func cleanEventTable() throws {
+    try EventHelper.cleanEventTable()
+  }
+  
+  fileprivate func cleanSpeechTable() throws {
+    try EventSpeechHelper.cleanSpeechTable()
+  }
+  
+  fileprivate func storeEvent() throws -> Identifier? {
+    return try EventHelper.storeEvent()
+  }
+  
+  fileprivate func fetchSpeeches(parameterId: Int) throws -> Response {
+    let req = Request.makeTest(method: .get)
+    try req.parameters.set("id", parameterId)
+    let res = try eventSpeechContoller.index(req: req).makeResponse()
+    return res
+  }
+  
+  fileprivate func storeSpeech(
+    forEventId eventId: Identifier,
```

Comment:
let's change `forEventId` to simple `for`  if you cary to inform user  about function parameters please remember user always can look at the function signature  by `alt+mouse click`

Reply:
fixed